activation/handler.go

```diff
@@ -241,7 +242,7 @@ func (h *Handler) SyntacticallyValidateDeps(
 		baseTickHeight = posAtx.TickHeight()
 	}
 
-	expectedChallengeHash := atx.NIPostChallenge.Hash()
+	expectedChallengeHash := wire.NIPostChallengeToWireV1(&atx.NIPostChallenge).Hash()
```

Comment:
Ideally the `SyntacticallyValidate` method accepts a `wire.ActivationTxV1` argument and not a `types.ActivationTx` (as well as all other methods used to handle incoming ATXs). Converting from wire to domain type should only be necessary once - after the ATX was successfully validated and just before it is stored in the DB.

Since the handler will always need to be able to validate V1 and V2 ATXs we will need validation methods for both kind of ATXs. I think the cleanest way to go here is to have the handler receive the raw bytes (from fetch or gossip) decode the first few bytes as pubepoch and then based on those call validation methods of the right (sub-) service that validate the specific ATX type for that public epoch.

Wdyt?

Reply:
Yeah. I'm currently refactoring the Handler and Validator methods for a follow up PR.

Reply:
Added some of the information from the discussion to this issue: https://github.com/spacemeshos/go-spacemesh/issues/5774